### DISCUSSION TITLE
DataSource golang-version

### DISCUSSION THREAD

**Discussion by pauldtill** [renovatebot/renovate#17726] [Request Help] [answered]
Hi,

We have a Dockerfile, which amongst other things, installs Go e.g.
```dockerfile
  wget https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz

```
I'd like to get this version managed by our self-hosted renovate bot (32.193.0).

I'd assumed the datasource - golang-version would handle this, but it's not quite working. The documentation doesn't provide any examples that I can see of configuring this datasource.

I've got this in the Dockerfile;
```dockerfile
# renovate: datasource=golang-version depName=golang extractVersion=^go(?<version>.*)$
ARG GO_VERSION=1.18.3
```

And the renovate debug logs show this;
```
DEBUG: Found no results from datasource that look like a version (repository=repo-name, dependency=golang)
       "result": {
         "homepage": "https://go.dev/",
         "sourceUrl": "https://github.com/golang/go",
         "releases": [],
         "registryUrl": "https://raw.githubusercontent.com/golang/website/"
       }

             "packageFile": "Dockerfile",
             "deps": [
               {
                 "depName": "golang",
                 "currentValue": "1.18.3",
                 "datasource": "golang-version",
                 "extractVersion": "^go(?<version>.*)$",
                 "replaceString": "# renovate: datasource=golang-version depName=golang extractVersion=^go(?<version>.*)$\nARG GO_VERSION=1.18.3\n",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "semver",
                 "sourceUrl": "https://github.com/golang/go",
                 "homepage": "https://go.dev/"
               },
```

We have a RegexManager configured with the below matchString, which is picking up the dependency OK and the current version, but the lookup is failing;
`"#\\srenovate:\\sdatasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?( extractVersion=(?<extractVersion>.*?))?\\s(ENV|ARG) .*?_VERSION=(?<currentValue>.*)\\s"`

What am I missing?

Thanks,
Paul

**Comment by Gabriel-Ladzaretti** [comment, marked as answer]
Try dropping the `extractVersion` entries as they are not needed when using the `golang-version` datasource.

Dockerfile
```diff
- # renovate: datasource=golang-version depName=golang extractVersion=^go(?<version>.*)$
+ # renovate: datasource=golang-version depName=golang 
```

renovate.json

```diff
- "#\\srenovate:\\sdatasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?( extractVersion=(?<extractVersion>.*?))?\\s(ENV|ARG) .*?_VERSION=(?<currentValue>.*)\\s"
+ "#\\srenovate:\\sdatasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s(ENV|ARG) .*?_VERSION=(?<currentValue>.*)\\s"
```

p.s. you probably should use `extractVersionTemplate` instead of the dockerfile embedded one anyways. with that said, it is still not needed in this case.

https://docs.renovatebot.com/configuration-options/#extractversiontemplate

**Reply by Gabriel-Ladzaretti** [reply]
reproduced your case here -
- https://github.com/ladzaretti/golang-regex-manger/pull/1

**Reply by pauldtill** [reply]
Brilliant - thank you, this has worked.
